src/phpdoc-parser/transpiler/README.md

```diff
@@ -0,0 +1,106 @@
+
+## Example Code Overview
+
+The example provided demonstrates how to parse a PHPDoc comment, extract node information, and transpile it into a TypeScript node structure. The main steps are as follows:
+
+1. **Tokenization**: Convert a PHPDoc comment string into tokens.
+2. **Parsing**: Transform the tokens into an Abstract Syntax Tree (AST) representing the PHPDoc structure.
+3. **Transpilation**: Convert the PHPDoc AST nodes into TypeScript type nodes.
+4. **Rendering**: Convert the TypeScript nodes into strings representing TypeScript type definitions.
+
+## Step 1: Import Necessary Classes and Types
+
+```TypeScript
+import { renderTsNodeToString } from './helpers';
+import { PhpDocTypeNodeToTypescriptTypeNodeTranspiler } from './php-doc-to-typescript-type-transpiler';
+import type { ParamTagValueNode, PropertyTagValueNode, ReturnTagValueNode } from '../ast/php-doc';
+import { Lexer, ConstExprParser, PhpDocParser, TokenIterator, TypeParser } from '../lexer/parser';
+```
+
+## Step 2: Define a Custom Transpiler Class
+
+```TypeScript
+class ExtendedTranspiler extends PhpDocTypeNodeToTypescriptTypeNodeTranspiler {
+  public customProperty: string = '';
+
+  constructor(public resolver: NameNodePathResolver<ExtendedTranspiler>) {
+    super((nodeParts: string[]) => resolver.call(this, nodeParts) as {
+      path: string;
+      name: string;
+      isTypeOnly: boolean;
+    });
+  }
+}
+```
+
+By extending the `PhpDocTypeNodeToTypescriptTypeNodeTranspiler` class to create a custom `ExtendedTranspiler` class, you gain the flexibility to add custom properties like `customProperty`. This allows you to tailor the transpiler's behavior to your project's needs. For example, the `customProperty` property can be used to determine how names are transformed or resolved within the `NameNodePathResolver`, catering to specific naming conventions with ease.
+
+## Step 3: Parse the PHPDoc Comment
+
+```TypeScript
+const transpileCommentText = (commentText: string) => {
+  // Initialize the parser objects.
+  const lexer = new Lexer();
+  const constExprParser = new ConstExprParser();
+  const typeParser = new TypeParser(constExprParser);
+  const phpDocParser = new PhpDocParser(typeParser, constExprParser);
+
+  // Tokenize and parse the PHPDoc comment to create the AST.
+  const tokens = new TokenIterator(lexer.tokenize(commentText));
+  const astRootNode = phpDocParser.parse(tokens); // Produces a PHPDocNode
+
+  // Extract property, return, or parameter nodes from the AST.
+  const propertyTagValueNodes = astRootNode
+    .getTags()
+    .map((node) => node.value)
+
+  return propertyTagValueNodes;
+};
+```
+
+## Step 4: Transpile PHPDoc Nodes to TypeScript and Render
+
+```TypeScript
+const commentText = `/**
+* @property-read  array|null  $person
+*/`;
+
+// Parse the PHPDoc comment text to get node structures.
+const tsTypeNode = transpileCommentText(commentText);
```

Comment:
I don't think tsTypeNode is a good variable name here.  
1. it is not a `ts` node so far 
2. it should be an array, I cannot see it by the name?